src/metadata.ts

```diff
@@ -0,0 +1,34 @@
+import * as constants from './lib/constants';
+import * as request from './lib/request';
+import { readFileSync, writeFileSync } from 'fs';
+import { reboot } from './device-state';
+import log from './lib/supervisor-console';
+
+export const initialized = (async () => {
+	if (constants.configJsonNonAtomicPath != null) {
+		const configJsonPath = constants.configJsonNonAtomicPath;
+		const config = JSON.parse(readFileSync(configJsonPath, 'utf-8'));
+		if (
+			!config.hasOwnProperty('unmanaged') &&
+			!config.hasOwnProperty('registered_at') &&
+			!config.hasOwnProperty('applicationId') &&
+			!config.hasOwnProperty('deviceId')
+		) {
+			const opts: request.requestLib.CoreOptions = {};
+
+			log.debug('Attempting to initialise from cloud metadata');
+			const url = 'http://169.254.169.254/latest/user-data';
```

Comment:
What is this IP?

Reply:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-add-user-data.html